backend/app/controllers/user.controller.js

```diff
@@ -1,25 +1,77 @@
+// const { User } = require("../models/user.model");
+const { user } = require("../models/index");
+var bcrypt = require("bcryptjs");
 
+exports.listUser = async (req, res) => {
+  try {
+    const users = await user.findAll();
+    res.json({
+      message: "User List",
+      data: users,
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: "Error",
+      error: error.message,
+    });
+  }
+};
+
+exports.getUser = async (req, res) => {
+  try {
+    const id = parseInt(req.params.id);
+
+    const temp = await user.findByPk(id);
 
-exports.listUser = (req, res) => {
-  res.json({
-    message: "List User",
-    data: []
-  });
+    res.json({
+      message: "User Detail",
+      data: temp,
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: "Error",
+      error: error.message,
+    });
+  }
 };
 
-exports.getUser = (req, res) => {
-  res.json({
-    message: "Get User",
-    data: {
-      id: req.params.id
+exports.deleteUser = async (req, res) => {
+  try {
+    const id = parseInt(req.params.id);
+    const temp = await user.findByPk(id);
+    await temp.destroy();
 
-    }
-  });
+    res.json({
+      message: "User Deleted",
+      data: temp,
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: "Error",
+      error: error.message,
+    });
+  }
 };
+exports.updateUser = async (req, res) => {
+  try {
+    const id = parseInt(req.params.id);
+    const { username, email, password, address } = req.body;
+    const temp = await user.findByPk(id);
+    temp.username = username;
+    temp.email = email;
+    temp.password =  bcrypt.hashSync(password, 8);
+    temp.address = address;
 
-exports.createUser = (req, res) => {
-  res.json({
-    message: "Create User",
-    data: req.body
-  });
-}
\ No newline at end of file
+    await temp.save();
+
+    res.json({
```

Comment:
Sử dụng http status code khác nhau cho từng loại api.

Ví dụ:

201 -> created
200 -> get/success
204 -> delete
404 -> not found
400 -> bad request.

Refer: https://www.npmjs.com/package/http-status-codes